fiftyone/core/dataset.py

```diff
@@ -52,6 +52,7 @@ def list_datasets():
         a list of :class:`Dataset` names
     """
     # pylint: disable=no-member
+    foo.start_db_service_if_necessary()
```

Comment:
I tried to avoid having to explicitly call `foo.start_db_service_if_necessary()` here by instead doing it anytime `Document.objects` is called (which requires metaclasses since that is a class method), but unfortunately the implementation below doesn't work because it seems to mess up inheritance order for other methods of MongoEngine documents...

```py
# In fiftyone.core.odm.document

from .database import start_db_service_if_necessary

class _MetaBaseDocument(mongoengine.base.TopLevelDocumentMetaclass):

    def __getattr__(self, name):
        if name == "objects":
            start_db_service_if_necessary()

        return super().__getattr__(name)

class BaseDocument(MongoEngineBaseDocument, metaclass=_MetaBaseDocument):
    ...
```

Reply:
I would think that using mongoengine classes cannot begin until `connect` is called, which expects the DB to have started? Regardless, putting the call inside mongoengine is probably overkill.

https://github.com/voxel51/fiftyone/blob/ae24ebc0a7ea22755020724735bbc7cd21631770/fiftyone/core/odm/database.py#L26

Reply:
I had to put `foo.start_db_service_if_necessary()` in the places that I did here because these are, indeed, times where `foo.DatasetDocument.objects` is accessed before any other DB connection is attempted.

I'm concerned that more such cases will appear over time (until we get rid of MongoEngine, I suppose) and lazy starting the DB as is will turn out to be a thorn more than a convenience...

I guess once we get rid of MongoEngine completely, then every DB operation would go through `connect()` and we'd be fine again. Maybe I should just replace `foo.DatasetDocument.objects` with pymongo now...